app/utility/RepDataMap.js

```diff
@@ -63,25 +63,35 @@ export const isValidData = (repData) => {
         // v3 bulk data flag check
         const bulkDataTransmissionFlag = parseData(repData, 21);
         console.tron.log("bulk data flag check at 21 " + bulkDataTransmissionFlag);
-        if (bulkDataTransmissionFlag == -9999) {
+        if (bulkDataTransmissionFlag == -9999 && !isInfinityorNegative(repData)) {
             return true;
         }
     } else if (initialFlag == -2345) {
         // v2 bulk data flag check
         const bulkDataTransmissionFlag = parseData(repData, 18);
         console.tron.log("bulk data flag check at 18 " + bulkDataTransmissionFlag);
-        if (bulkDataTransmissionFlag == -9999) {
+        if (bulkDataTransmissionFlag == -9999 && !isInfinityorNegative(repData)) {
             return true;
         }
     } else if (initialFlag == -1234) {
         // v1 bulk data flag check
         const bulkDataTransmissionFlag = parseData(repData, 6);
         console.tron.log("bulk data flag check at 6 " + bulkDataTransmissionFlag);
-        if (bulkDataTransmissionFlag == -9999) {
+        if (bulkDataTransmissionFlag == -9999 && !isInfinityorNegative(repData)) {
             return true;
         }
     }
 
     // default
     return false;
 };
+
+const isInfinityorNegative = (repData) => {
+    let averageVelocity = RepDataMap.averageVelocity(repData);
+
+    if (!isFinite(averageVelocity) && averageVelocity < 0) {
```

Comment:
- [ ] Wait isn't this the wrong logic? Shouldn't it be or? We might need tests if there are this many mistakes